docs/DeveloperGuide.md

```diff
@@ -2,28 +2,131 @@
+4. Adds the new `Group` object to the `groups` list.
+5. Returns the newly created `Group` object.
+6. If the group already exists, prints an error message indicating that the group already exists.
+
+### Add Member to Group feature
+
+#### Implementation
+
+The "Add Member to Group" feature is facilitated by the `Group` class. It provides methods to manage group membership
+and allows users to add new members to an existing group. The implementation of this feature is as follows:
+
+The Group class maintains a list of members as a `private List<User>` field called `members`.
+
+Users can add a new member to the group by using the command `member USER_NAME`. The `addMember(String memberName)` method is responsible for adding a new member to the group. It performs the following steps:
+
+1. Checks if a user with the given `memberName` is already a member of the group using the `isMember(String memberName)`
+   method.
+2. If the user is not a member, creates a new `User` object with the provided `memberName`.
+3. Adds the new `User` object to the `members` list.
+4. Prints a success message indicating that the member has been added to the group.
+
+![Sequence Diagram](AddMember.png)
+
+### Expenses feature
+
+#### Implementation
+
+The Expenses feature is facilitated through the Expense class. It allows users to add a new Expense through creation of
+a new Expense object. Users can specify amount paid, the payee, and the members of the group involved in the
+transaction.
+Additionally, it implements the following operations:
+
++ `Expenses#payer()` - Gives the name of the member who paid for the expense
++ `Expenses#totalAmount()` - Returns the total amount of the expense
++ `Expenses#payees()` - Returns all the members involved in the transaction
+
+These operations are exposed in the Expense class through the `getPayerName()`, `getTotalAmount()`, and `getPayees()`
+functions respectively.
+
+### Balance feature
+
+#### Implementation
+
+The Balance feature is facilitated through the Balance class.
+It allows a user to view a printed list of other users in the Group, and the amount that is owed by/to each user.
+
+Each `Balance` object contains a String of a user `userName`,
+and a Map `balanceList`. This Map uses String of other users' usernames as Key, and
+a Float of the amount that is owed by/to each user.
+
+To print a user's Balance List, perform the following steps:
+
+1. Create a `Balance` object with the params String `userName` and the current Group `group`.
+2. From the `members` and `expenseList` List items in `group`, the Map `balanceList` is populated.
+3. Call method `printBalance()` to print the contents of the Map `balanceList`.
+
+### Settle feature
+
+#### Implementation
+
+The Settle feature is facilitated through the Settle class.
+It allows a user to settle the debts between two users in a Group.
+
+ The `Settle` class contains a `settleDebt(String userName1, String userName2)` method.
```

Comment:
Nice feature! However, the Settle class does not include a settleDebt( ) method that takes in 2 string inputs. It would be helpful to either include this method in the Settle class or clarify where it is implemented.